### PR TITLE
feat: Quick Order Screen Reader improvements (#13654)

### LINKED ISSUE #13654: Quick Order: Accessibility Pattern revision
**Describe the bug**
- Although there is no keyboard trap, the Accessibility Pattern for this UI needs revision. Currently, the search box is triggered on "Enter" instead of using a button. Extra descriptions for screen reader users will be needed to convey functionality. Just to make sure this is what we want.
- There needs to be consistency among all internal search boxes so implementation is similar in all cases. E.g. Find a Store uses a button to trigger the search.
- Currently, all search boxes have different interaction patterns (product search, find a store, quick order), e.g. trigger mechanism, clear button, autocomplete
- This was already brought to the attention of @Manurod2018 and @vekterdude 

Screen Shot 2021-09-08 at 4 25 48 PM

### PR DESCRIPTION
**Pull request by dydome**
close #13654

### CODE REVIEW

**Review by developpeurweb — CHANGES_REQUESTED** [review]
The alert and the input field hint work well for all SR.
Now, for some reason, the search query highlight stays as part of the `alt` attribute, it would be ideal to clean it up, if not possible, then in this case, since the image is NOT-a-link, it's OK to set it up as `role="presentation"` so it gets ignored by the SR. We have the product name being a link and also focusable therefore `role="presentation"` it's fine, if no other solution can be found.
Screen Shot 2022-01-24 at 1 32 27 PM

**Comment by dydome** [comment]
@developpeurweb I have changed alt content to the product name and also added role="presentation" as it is good 
 way I think.

File: feature-libs/cart/quick-order/components/quick-order/table/item/quick-order-item.component.html [hunk 1]
```diff
@@ -9,8 +9,9 @@
         >
           <cx-media
             [container]="entry?.product?.images?.PRIMARY"
-            [alt]="entry.product.summary ?? entry.product.name"
+            [alt]="entry?.product?.name"
             format="cartIcon"
+            role="presentation"
```
Comment (developpeurweb) [inline feature-libs/cart/quick-order/components/quick-order/table/item/quick-order-item.component.html]: Seems the `alt` attribute is clean and works well now, so please remove `role="presentation"` as both cannot coexist. Their purposes contradict each other.

**Review by developpeurweb — CHANGES_REQUESTED** [review]

File: feature-libs/cart/quick-order/components/quick-order/table/item/quick-order-item.component.html (hunk 1 above)
Comment (dydome) [reply feature-libs/cart/quick-order/components/quick-order/table/item/quick-order-item.component.html]: Removed

**Review by developpeurweb — APPROVED** [review]

**Pull request merged**